plugins/java-decompiler/engine/src/org/jetbrains/java/decompiler/modules/decompiler/exps/ConstExprent.java

```diff
@@ -31,6 +31,67 @@
+  static {
+    final double PI_D = Math.PI;
+    final float PI_F = (float)Math.PI;
+    PI_DOUBLES.put(PI_D, new String[] { "", "" });
+    PI_DOUBLES.put(-PI_D, new String[] { "-", "" });
+    PI_FLOATS.put(PI_F, new String[] { "", "" });
+    PI_FLOATS.put(-PI_F, new String[] { "-", "" });
+
+    PI_DOUBLES.put(PI_D * 2D, new String[] { "(", " * 2D)" });
+    PI_DOUBLES.put(-PI_D * 2D, new String[] { "(-", " * 2D)" });
+    PI_FLOATS.put(PI_F * 2F, new String[] { "(", " * 2F)" });
+    PI_FLOATS.put(-PI_F * 2F, new String[] { "(-", " * 2F)" });
+
+    PI_DOUBLES.put(PI_D / 2D, new String[] { "(", " / 2D)" });
+    PI_DOUBLES.put(-PI_D / 2D, new String[] { "(-", " / 2D)" });
+    PI_FLOATS.put(PI_F / 2F, new String[] { "(", " / 2F)" });
+    PI_FLOATS.put(-PI_F / 2F, new String[] { "(-", " / 2F)" });
+
+    PI_DOUBLES.put(PI_D * 1.5D, new String[] { "(", " * 1.5D)" });
+    PI_DOUBLES.put(-PI_D * 1.5D, new String[] { "(-", " * 1.5D)" });
+    PI_FLOATS.put(PI_F * 1.5F, new String[] { "(", " * 1.5F)" });
+    PI_FLOATS.put(-PI_F * 1.5F, new String[] { "(-", " * 1.5F)" });
+
+    PI_DOUBLES.put(PI_D / 3D, new String[] { "(", " / 3D)" });
+    PI_DOUBLES.put(-PI_D / 3D, new String[] { "(-", " / 3D)" });
+    PI_FLOATS.put(PI_F / 3F, new String[] { "(", " / 3F)" });
+    PI_FLOATS.put(-PI_F / 3F, new String[] { "(-", " / 3F)" });
+
+    PI_DOUBLES.put(PI_D / 4D, new String[] { "(", " / 4D)" });
+    PI_DOUBLES.put(-PI_D / 4D, new String[] { "(-", " / 4D)" });
+    PI_FLOATS.put(PI_F / 4F, new String[] { "(", " / 4F)" });
+    PI_FLOATS.put(-PI_F / 4F, new String[] { "(-", " / 4F)" });
+
+    PI_DOUBLES.put(PI_D / 5D, new String[] { "(", " / 5D)" });
+    PI_DOUBLES.put(-PI_D / 5D, new String[] { "(-", " / 5D)" });
+    PI_FLOATS.put(PI_F / 5F, new String[] { "(", " / 5F)" });
+    PI_FLOATS.put(-PI_F / 5F, new String[] { "(-", " / 5F)" });
+
+    PI_DOUBLES.put(PI_D / 6D, new String[] { "(", " / 6D)" });
+    PI_DOUBLES.put(-PI_D / 6D, new String[] { "(-", " / 6D)" });
+    PI_FLOATS.put(PI_F / 6F, new String[] { "(", " / 6F)" });
+    PI_FLOATS.put(-PI_F / 6F, new String[] { "(-", " / 6F)" });
+
+    PI_DOUBLES.put(PI_D / 8D, new String[] { "(", " / 8D)" });
+    PI_DOUBLES.put(-PI_D / 8D, new String[] { "(-", " / 8D)" });
+    PI_FLOATS.put(PI_F / 8F, new String[] { "(", " / 8F)" });
+    PI_FLOATS.put(-PI_F / 8F, new String[] { "(-", " / 8F)" });
+
+    PI_DOUBLES.put(PI_D / 10D, new String[] { "(", " / 10D)" });
+    PI_DOUBLES.put(-PI_D / 10D, new String[] { "(-", " / 10D)" });
+    PI_FLOATS.put(PI_F / 10F, new String[] { "(", " / 10F)" });
+    PI_FLOATS.put(-PI_F / 10F, new String[] { "(-", " / 10F)" });
+
+    // Radian/degree conversions
+    PI_DOUBLES.put(PI_D / 180D, new String[] { "(", " / 180D)" });
+    PI_DOUBLES.put(180D / PI_D, new String[] { "(180D / ", ")" });
+    PI_FLOATS.put(PI_F / 180F, new String[] { "(", " / 180F)" });
+    PI_FLOATS.put(180F / PI_F, new String[] { "(180F / ", ")" });
+  }
```

Comment:
I don't think deciphering these expressions significantly improves readability of a decompiled code, especially since it leaves out some constants anyway.

Reply:
I'd want to keep at least `PI / 2`, `PI * 2`, and the radian-degree ones as they're pretty helpful for me and come up in trig-related code (for instance in MC `57.2957763671875D` is in a number of places which is really hard to recognize, but happens to be `(double)(180F / (float)Math.PI))`).

It might be possible to try and identify any rational number (or rational number multiplied by pi) using [Farey sequences](https://en.wikipedia.org/wiki/Farey_sequence#Applications), though I don't know if that's worthwhile.

Reply:
I see. Still, I do believe it would be more accurate to leave values as-is (except exact `Math.PI` and `Math.E` values), and provide possible interpretation in a comment block. The same applies for float approximations.